src/Testing/MakesGraphQLRequests.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace Nuwave\Lighthouse\Testing;
+
+use Illuminate\Foundation\Testing\TestResponse;
+
+/**
+ * Useful helpers for PHPUnit testing.
+ *
+ * It depends upon methods defined in
+ * @see \Illuminate\Foundation\Testing\Concerns\MakesHttpRequests
+ */
+trait MakesGraphQLRequests
+{
+    /**
+     * Visit the given URI with a POST request, expecting a JSON response.
+     *
+     * @param  string  $uri
+     * @param  array  $data
+     * @param  array  $headers
+     * @return \Illuminate\Foundation\Testing\TestResponse
+     */
+    abstract public function postJson($uri, array $data = [], array $headers = []);
+
+    /**
+     * Call the given URI and return the Response.
+     *
+     * @param  string  $method
+     * @param  string  $uri
+     * @param  array  $parameters
+     * @param  array  $cookies
+     * @param  array  $files
+     * @param  array  $server
+     * @param  string  $content
+     * @return \Illuminate\Foundation\Testing\TestResponse
+     */
+    abstract public function call($method, $uri, $parameters = [], $cookies = [], $files = [], $server = [], $content = null);
+
+    /**
+     * Transform headers array to array of $_SERVER vars with HTTP_* format.
+     *
+     * @param  array  $headers
+     * @return array
+     */
+    abstract protected function transformHeadersToServerVars(array $headers);
+
+    /**
+     * Execute a query as if it was sent as a request to the server.
+     *
+     * @param  string  $query
+     * @return \Illuminate\Foundation\Testing\TestResponse
+     */
+    protected function queryGraphQL(string $query): TestResponse
```

Comment:
Could we make this smarter or add another method for batched queries?
Right now the only way to send batched queries is to do 

```
$this->postGraphQL([
            [
                'query' => '
                    {
                        user {
                            email
                        }
                    }
                    ',
            ],
            [
                'query' => '
                    {
                        user {
                            name
                        }
                    }
                    ',
            ],
        ])
```

I think it makes sense to either make queryGraphQL take in a array of query strings and then it adds the `query` key in front of all the values, or make another method called something like `queryGraphQLBatched`.

Reply:
The example call for batched queries seems fine to me. I don't think that testing through batched queries is very common.

Reply:
My thoughts was just to remove the need for adding query, so I could write 

```
$this->postGraphQL([
    '{
        user {
            email
        }
    }',
    '{
        user {
            name
        }
    }'
])
```

But I guess you are right that not many will use that feature, and they can easily create the helper if they think the other syntax is too long.